src/dosinst.c

```diff
@@ -505,7 +505,7 @@ uninstall_check(int skip_question)
     HKEY	uninstall_key_handle;
     char	*uninstall_key = "software\\Microsoft\\Windows\\CurrentVersion\\Uninstall";
     char	subkey_name_buff[BUFSIZE];
-    char	temp_string_buffer[BUFSIZE];
+    char	temp_string_buffer[BUFSIZE-2];
     DWORD	local_bufsize = BUFSIZE;
```

Comment:
You changed the size of `temp_string_buffer`, then shouldn't this line be also `BUFSIZE-2` (or `sizeof(temp_string_buffer)`)?
Also the same in line 529, 541 and 573.